internal/elasticsearch/logstash/pipeline.go

```diff
@@ -0,0 +1,174 @@
+package logstash
+
+import (
+	"context"
+	"time"
+
+	"github.com/elastic/terraform-provider-elasticstack/internal/clients"
+	"github.com/elastic/terraform-provider-elasticstack/internal/models"
+	"github.com/elastic/terraform-provider-elasticstack/internal/utils"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func ResourceLogstashPipeline() *schema.Resource {
+	logstashPipelineSchema := map[string]*schema.Schema{
+		"id": {
+			Description: "Internal identifier of the resource",
+			Type:        schema.TypeString,
+			Computed:    true,
+		},
+		"pipeline_id": {
+			Description: "Identifier for the pipeline.",
+			Type:        schema.TypeString,
+			Required:    true,
+			ForceNew:    true,
+		},
+		"description": {
+			Description: "Description of the pipeline.",
+			Type:        schema.TypeString,
+			Optional:    true,
+		},
+		"last_modified": {
+			Description: "Date the pipeline was last updated.",
+			Type:        schema.TypeString,
+			Computed:    true,
+		},
+		"pipeline": {
+			Description: "Configuration for the pipeline.",
+			Type:        schema.TypeString,
+			Required:    true,
+		},
+		"pipeline_metadata": {
+			Description: "Optional metadata about the pipeline.",
+			Type:        schema.TypeMap,
+			Optional:    true,
+			Elem: &schema.Schema{
+				Type:    schema.TypeString,
+				Default: nil,
+			},
+		},
+		"pipeline_settings": {
```

Comment:
IMO it would be nicer to have explicit schema items for the handful of `pipeline.*` available in [Logstash](https://www.elastic.co/guide/en/logstash/8.4/logstash-settings-file.html). We'd be able to validate individual settings (both keys and values) within the provider rather than simply allow a free form map. 

It would also be consistent with how we're now treating [index](https://github.com/elastic/terraform-provider-elasticstack/pull/137) settings. 

WDYT?

Reply:
I actually started attempting to do this before throwing my toys out of the pram and going with the simpler approach, will take another look at this today though to see how to achieve this!

Reply:
Bit more testing and maybe a few tweaks left, nearly got it working, just some weird behaviour surrounding defaults when not defining the pipeline_settings block in the resource. Will take another look tomorrow 👍

Reply:
@tobio - My brain clearly wasn't working the other day as I was building based on the deprecated approach in index settings... apologies for this! 

I have just updated the approach to follow the newer one in https://github.com/elastic/terraform-provider-elasticstack/pull/151/commits/1a45bda9bb1e71522b5c3f902f0505e7b38c6d38, this has also solved the issue I was having with the defaults not being applied when `pipeline_settings {}` was not set, so win win!

Reply:
Exactly what I was looking to test. Thanks for this!